### DISCUSSION TITLE
What is the recommended way to disable local filesystem access?

### DISCUSSION THREAD

**Discussion by nathanielc** [apache/datafusion#12636] [Q&A] [answered]
I am exposing a datafusion context via FlightSQL and I would like to prevent queries from reading and writing data to the local filesystem.

I see a few approaches...

### Empty ObjectStoreRegistry 

I could implement the ObjectStoreRegistry and return an error for all local urls.

### Replace ObjectStore for local urls

I could replace the default ObjectStore for the local urls with an implementation of the ObjectStore that returns not found/not supported for methods

### Something Else?

Thanks

**Comment by nathanielc** [comment]
Or is the recommended way to simply rely on the [SQLOptions](https://docs.rs/datafusion/latest/datafusion/execution/context/struct.SQLOptions.html)? Is the intent that if a query cannot create a new table pointing to the local filesystem they cannot read/write to it?

**Reply by alamb** [reply, marked as answer]
That is how we do it in InfluxDB

> I could replace the default ObjectStore for the local urls with an implementation of the ObjectStore that returns not found/not supported for methods



This sounds like a good idea too
